binaries/build.py

```diff
@@ -62,30 +57,34 @@ def build(args):
 
     os.chdir(REPO_ROOT)
 
-    ts_wheel_path = glob.glob(os.path.join(REPO_ROOT, "dist", "*.whl"))[0]
-    ma_wheel_path = glob.glob(
-        os.path.join(REPO_ROOT, "model-archiver", "dist", "*.whl")
-    )[0]
-    wa_wheel_path = glob.glob(
-        os.path.join(REPO_ROOT, "workflow-archiver", "dist", "*.whl")
-    )[0]
+    if not args.dry_run:
+        ts_wheel_path = glob.glob(os.path.join(REPO_ROOT, "dist", "*.whl"))[0]
+        ma_wheel_path = glob.glob(
+            os.path.join(REPO_ROOT, "model-archiver", "dist", "*.whl")
+        )[0]
+        wa_wheel_path = glob.glob(
+            os.path.join(REPO_ROOT, "workflow-archiver", "dist", "*.whl")
+        )[0]
+
+    else:
+        ts_wheel_path = os.path.join(REPO_ROOT, "dist", "*.whl")
+        ma_wheel_path = os.path.join("model-archiver", "dist", "*.whl")
+        wa_wheel_path = os.path.join("workflow-archiver", "dist", "*.whl")
+
     print(f"## TorchServe wheel location: {ts_wheel_path}")
     print(f"## Model archiver wheel location: {ma_wheel_path}")
     print(f"## Workflow archiver wheel location: {ma_wheel_path}")
 
     # Build TS & MA on Conda if available
     conda_build_exit_code = 0
     if not is_conda_env():
-        install_miniconda()
+        install_miniconda(args.dry_run)
 
     if not is_conda_build_env():
-        install_conda_build()
+        install_conda_build(args.dry_run)
 
     conda_build_exit_code = conda_build(
-        ts_wheel_path,
-        ma_wheel_path,
-        wa_wheel_path,
-        args.nightly,
+        ts_wheel_path, ma_wheel_path, wa_wheel_path, args.nightly, args.dry_run
```

Comment:
Suggestion: I wonder if it makes sense to pass args , instead of passing them individually

Reply:
I don't have a strong opinion, either way they need to be unpacked in the calling function so it's not like we save that much code